morebits.js

```diff
@@ -1152,6 +1152,72 @@ Morebits.array = {
 	}
 };
 
+/**
+ * ************ Morebits.select2 ***************
+ * Utilities to enhance select2 menus
+ * See twinklewarn and twinklexfd for sample usages
+ */
+Morebits.select2 = {
+
+	/**
+	 * Custom matcher in which if the optgroup matches, all options in that group are shown,
+	 * like in jquery.chosen
+	 */
+	matcher: function(params, data) {
+		var originalMatcher = $.fn.select2.defaults.defaults.matcher;
+		var result = originalMatcher(params, data);
+
+		if (result && params.term &&
+			data.text.toUpperCase().indexOf(params.term.toUpperCase()) !== -1) {
+			result.children = data.children;
+		}
+		return result;
+	},
+
+	/** Underline matched part of options */
+	highlightSearchMatches: function(data) {
```

Comment:
I said this already, @siddharthvp, but these really are quite well done.  Way more concise than I ever would have thought possible.

Reply:
The idea comes from https://forums.select2.org/t/how-can-i-highlight-the-results-on-a-search/52/2. I just condensed down the code. 

That should also explain the `queryInterceptor` (they've used the implicitly global `query` variable). Too bad that `templateResult` doesn't expose the search query as a parameter. This hack is unlikely to break, unless select2 makes some deliberate breaking changes.